### PR TITLE
<WIP> Add launch file to bringup nav2 with namespace

### PR DESCRIPTION
**Pull request by lbegani**
DO NOT MERGE

This change is to support multi robot system.

---

## Basic Info

| Info | Please fill out this column |
| ------ | ----------- |
| Ticket(s) this addresses   | #1366  |
| Primary OS tested on | Ubuntu18.04 |
| Robotic platform tested on | TB3 hardware |

---

## Description of contribution in a few bullet points
* Create a bringup launch file with namespace support

---

## Future work that may be required in bullet points
* Remove some hardcodings that still exists in the launch file. Need to extract the string from namespace arg and use it for param/rviz etc file selection.
* Reorg the launch files currently used for multi-robot simulation so that common files are used for multi-robot setup in simulation as well as for hardware.

### CODE REVIEW

File: nav2_bringup/bringup/launch/nav2_bringup_namespace_launch.py [hunk 1]
```diff
@@ -0,0 +1,201 @@
...
+# limitations under the License.
+
+"""This is all-in-one launch script intended for use by nav2 developers."""
+
+import os
+
+from ament_index_python.packages import get_package_share_directory
+
+from launch import LaunchDescription
+from launch.actions import (DeclareLaunchArgument, EmitEvent, ExecuteProcess,
```
Comment (orduno) [inline nav2_bringup/bringup/launch/nav2_bringup_namespace_launch.py]: Not using EmitEvent, RegisterEventHandler, ExecuteProcess, can remove

File: nav2_bringup/bringup/launch/nav2_bringup_namespace_launch.py [hunk 2]
```diff
@@ -0,0 +1,201 @@
...
+                            IncludeLaunchDescription, RegisterEventHandler,
+                            GroupAction)
+from launch.conditions import IfCondition
+from launch.event_handlers import OnProcessExit
```
Comment (orduno) [inline nav2_bringup/bringup/launch/nav2_bringup_namespace_launch.py]: Remove OnProcessExit

File: nav2_bringup/bringup/launch/nav2_bringup_namespace_launch.py [hunk 3]
```diff
@@ -0,0 +1,201 @@
...
+from launch.events import Shutdown
```
Comment (orduno) [inline nav2_bringup/bringup/launch/nav2_bringup_namespace_launch.py]: Remove OnShutdown

File: nav2_bringup/bringup/launch/nav2_bringup_namespace_launch.py [hunk 4]
```diff
@@ -0,0 +1,201 @@
...
+    ld.add_action(declare_use_remappings_cmd)
+
+    ld.add_action(declare_rviz_config_file_cmd)
+    ld.add_action(declare_use_simulator_cmd)
+    ld.add_action(declare_use_robot_state_pub_cmd)
+    ld.add_action(declare_use_rviz_cmd)
+    ld.add_action(declare_simulator_cmd)
+    ld.add_action(declare_world_cmd)
+
+    # Add any conditioned actions
```
Comment (orduno) [inline nav2_bringup/bringup/launch/nav2_bringup_namespace_launch.py]: Remove commented code.

File: nav2_bringup/bringup/launch/nav2_bringup_namespace_launch.py [hunk 5]
```diff
@@ -0,0 +1,201 @@
...
+        'use_rviz',
+        default_value='True',
+        description='Whether to start RVIZ')
+
+    declare_simulator_cmd = DeclareLaunchArgument(
+        'simulator',
+        default_value='gazebo',
+        description='The simulator to use (gazebo or gzserver)')
+
+    declare_world_cmd = DeclareLaunchArgument(
```
Comment (orduno) [inline nav2_bringup/bringup/launch/nav2_bringup_namespace_launch.py]: Don't need this launch argument.

File: nav2_bringup/bringup/launch/nav2_bringup_namespace_launch.py [hunk 6]
```diff
@@ -0,0 +1,201 @@
...
+    map_yaml_file = LaunchConfiguration('map')
+    use_sim_time = LaunchConfiguration('use_sim_time')
+    params_file = LaunchConfiguration('params_file')
+    bt_xml_file = LaunchConfiguration('bt_xml_file')
+    autostart = LaunchConfiguration('autostart')
+    use_remappings = LaunchConfiguration('use_remappings')
+
+    # Launch configuration variables specific to simulation
+    rviz_config_file = LaunchConfiguration('rviz_config_file')
+    use_simulator = LaunchConfiguration('use_simulator')
```
Comment (orduno) [inline nav2_bringup/bringup/launch/nav2_bringup_namespace_launch.py]: We don't need `use_simulator`, `use_robot_state_pub`, `simulator` and `world`.

**Comment by SteveMacenski** [comment]
@lbegani @orduno what's the status of this?

**Comment by orduno** [comment]
We can close it as it was resolved by #1446.

**Pull request closed without merging**